hal/halx86/apic/apic.c

```diff
@@ -94,6 +94,7 @@ ULONG
 IOApicRead(UCHAR Register)
 {
     /* Select the register, then do the read */
+    ASSERT(Register <= 0x3F);
     *(volatile UCHAR *)(IOAPIC_BASE + IOAPIC_IOREGSEL) = Register;
```

Comment:
```suggestion
    *(volatile ULONG *)(IOAPIC_BASE + IOAPIC_IOREGSEL) = Register;
```

Reply:
Alternatively,
```suggestion
    WRITE_REGISTER_ULONG(IOAPIC_BASE + IOAPIC_IOREGSEL, Register);
```

Reply:
I heard one does not simply `xxx_REGISTER_xxx` or `xxx_PORT_xxx` inside critical parts of HAL :ok_hand:

Reply:
I've heard that when code uses the correct header, these become inlined macros that do the correct thing.
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/ddk/ioaccess.h;hb=46c84779b6b07f1b9a172c6b357eba458770d1a2#l36